src/Database/LSMTree/Internal/Serialise/Class.hs

```diff
@@ -87,10 +100,27 @@ class SerialiseValue v where
 serialiseValueIdentity :: (Eq v, SerialiseValue v) => v -> Bool
 serialiseValueIdentity x = deserialiseValue (serialiseValue x) == x
 
+-- | Test the __Identity up to slicing__ law for the 'SerialiseValue' class
+serialiseValueIdentityUpToSlicing ::
+     (Eq v, SerialiseValue v)
+  => RawBytes -> v -> RawBytes -> Bool
+serialiseValueIdentityUpToSlicing prefix x suffix =
+    deserialiseValue (packSlice prefix (serialiseValue x) suffix) == x
+
 -- | Test the __Concat distributes__ law for the 'SerialiseValue' class
 serialiseValueConcatDistributes :: forall v. (Eq v, SerialiseValue v) => Proxy v -> [RawBytes] -> Bool
 serialiseValueConcatDistributes _ xs = deserialiseValueN @v xs == deserialiseValue (mconcat xs)
 
+{-------------------------------------------------------------------------------
+  RawBytes
+-------------------------------------------------------------------------------}
+
+-- | @'packSlice' prefix x suffix@ makes @x@ into a slice with @prefix@ bytes on
+-- the left and @suffix@ bytes on the right.
+packSlice :: RawBytes -> RawBytes -> RawBytes -> RawBytes
+packSlice prefix x suffix =
+    RB.take (RB.size x) (RB.drop (RB.size prefix) (prefix <> x <> suffix))
```

Comment:
This function feels a bit out of place here, only ever useful to state the property. But it makes sense that you want to check that instances handle slicing correctly and I don't see a nicer way of phrasing the property, so I'm on board.

Reply:
I agree it feels slightly out of place